### PR TITLE
Importing The Same JDL File Multiple Times Will Lead to Duplicated Menus Items

### LINKED ISSUE #15745: Importing The Same JDL File Multiple Times Will Lead to Duplicated Menus Items With The Vue Front End
##### **Overview of the issue**

After importing the same JDL file a second time, I notice that there are duplicated menu items under the Entities menu. After importing the same JDL file one more time, there are a third set of menu items. 

##### **Reproduce the error**

Create a JDL file with at least one entity. Import the file multiple times. Bring up the app after each import and check the Entities menu.

##### **JHipster configuration**

```
{
  "generator-jhipster": {
    "applicationType": "monolith",
    "baseName": "myapp",
    "jhipsterVersion": "7.1.0",
    "skipClient": false,
    "skipServer": false,
    "skipUserManagement": false,
    "skipCheckLengthOfIdentifier": false,
    "skipFakeData": false,
    "jhiPrefix": "jhi",
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "testFrameworks": ["cypress", "gatling"],
    "blueprints": [],
    "otherModules": [],
    "pages": [],
    "creationTimestamp": 1625095262715,
    "serviceDiscoveryType": false,
    "reactive": false,
    "authenticationType": "jwt",
    "packageName": "com.abc.myapp",
    "serverPort": "8080",
    "cacheProvider": "ehcache",
    "enableHibernateCache": true,
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "postgresql",
    "buildTool": "gradle",
    "serverSideOptions": [],
    "websocket": false,
    "searchEngine": false,
    "messageBroker": false,
    "enableSwaggerCodegen": false,
    "clientFramework": "vue",
    "withAdminUi": true,
    "clientTheme": "cosmo",
    "clientThemeVariant": "primary",
    "enableTranslation": true,
    "nativeLanguage": "en",
    "packageFolder": "com/abc/myapp",
    "jwtSecretKey": "NThkZDg5YjE3OTkxNGU2YjUzNjAwODMwZjY4ODRiYmIyOWVkN2Q0NmZhMjZhNDU4MmQyYTQ5NDcxOWZlODRjY2ZhYWE5NjNjZWJkZDUyYzljYTU0NmU0OGExZDY4N2EwZjljZWQwMTliYWU3NGZiZTQ3NDQ1MGIwOTNjZDU3NmE=",
    "devServerPort": 9060,
    "clientPackageManager": "npm",
    "languages": ["en", "es"]
  }
}
```

##### **Browsers and Operating System**

- [ ] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by akang**
Importing The Same JDL File Multiple Times Will Lead to Duplicated Menus Items

Fix #15745

### CODE REVIEW

**Comment by akang** [comment]
Hi all,

First PR for this project. Just wondering if I have ticked all the right boxes as far as the PR submission is concerned?

thanks
aka

File: utils/field.js [hunk 1]
```diff
@@ -98,6 +99,19 @@ const generateFakeDataForField = (field, faker, changelogDate, type = 'csv') =>
...
+    } catch (e) {
+      console.log(
+        `${chalk.yellowBright('WARNING!')} ${field.fieldName} pattern is not valid: ${
+          field.fieldValidateRulesPattern
+        }. Skipping generating fake data. `
+      );
+      return undefined;
+    }
+
     const generated = field.createRandexp().gen();
```
Comment (mshima) [inline utils/field.js:115]:
```suggestion
    const dataGenerator = field.createRandexp();
    if (!dataGenerator) {
      return undefined;
    }
    const generated = dataGenerator.gen();
```

- Move the regex check to:
https://github.com/jhipster/generator-jhipster/blob/cbd88e4daa6681c2a512c8a73c6621e156564638/utils/field.js#L374
like
```
    field.createRandexp = () => {
      // check field.fieldValidateRulesPattern and return undefined.
      return faker.createRandexp(field.fieldValidateRulesPattern);
    }
```

- Replace console.log with this.warning.

**Review by mshima — COMMENTED** [review]
In angular framework, this is the test that check needles duplication:
https://github.com/jhipster/generator-jhipster/blob/4fec42623b8e7608017e854ac2df565e6c6861a9/test/needle-api/needle-client-angular.spec.js#L69-L75

Would be nice to add to vue, but not required.

File: utils/field.js (hunk 1 above)
Comment (akang) [reply utils/field.js]: Done

File: utils/field.js [hunk 2]
```diff
@@ -19,6 +19,7 @@
 
 const assert = require('assert');
 const _ = require('lodash');
+const chalk = require('chalk');
```
Comment (mshima) [inline utils/field.js:22]:
```suggestion
```

**Comment by akang** [comment]
I used the wrong email when pushing the codes and now the "license/cla" agreement is unsigned. 

@mshima is there a workaround to get this task ticked off?

**Comment by mshima** [comment]
@akang you used 2 different git users (different email) to create the commit.
You should use only the one with the email you signed the cla, otherwise it won’t become green.

**Review by mshima — APPROVED** [review]

**Pull request merged**